components/brave_shields/browser/ad_block_service.cc

```diff
@@ -96,7 +96,11 @@ AdBlockService::SourceProviderObserver::~SourceProviderObserver() {
   resource_provider_->RemoveObserver(this);
 }
 
-void AdBlockService::SourceProviderObserver::OnChanged() {
+void AdBlockService::SourceProviderObserver::OnChanged(bool is_default_engine) {
+  if (adblock_engine_->IsDefaultEngine() && !is_default_engine) {
```

Comment:
@antonok-edm @bridiver Is it true that we have to rebuild the additional engine if the default is updated?
It doesn't clear to me, but the comments tell:
`The regional adblock engines depend on the default engine for resource loads.`
([link](https://github.com/brave/brave-core/blob/fix-adblock-ruleset-reparsing/browser/brave_shields/ad_block_service_browsertest.cc#L276-L277))

The best option to put here `adblock_engine_->IsDefaultEngine() != !is_default_engine`

Reply:
afaik the two engines are independent and we should not have to reload the "extras" engine when the default changes or vice versa

Reply:
> `The regional adblock engines depend on the default engine for resource loads.`

That comment should be removed; that isn't the case anymore. @bridiver is correct.

Reply:
should we do this instead?
```cpp
if (adblock_engine_->IsDefaultEngine() != is_default_engine) {
```

That way we can skip updates of the default engine for the additional engine as well.

Reply:
Changed.